src/mixer/basetrackplayer.cpp

```diff
@@ -343,22 +343,44 @@ void BaseTrackPlayerImpl::slotTrackLoaded(TrackPointer pNewTrack,
                 ConfigKey("[Mixer Profile]", "GainAutoReset"), false)) {
             m_pPreGain->set(1.0);
         }
-        int reset = m_pConfig->getValue<int>(
-                ConfigKey("[Controls]", "SpeedAutoReset"), RESET_PITCH);
-        if (reset == RESET_SPEED || reset == RESET_PITCH_AND_SPEED) {
-            // Avoid resetting speed if master sync is enabled and other decks with sync enabled
-            // are playing, as this would change the speed of already playing decks.
-            if (!m_pEngineMaster->getEngineSync()->otherSyncedPlaying(getGroup())) {
-                if (m_pRateSlider != NULL) {
-                    m_pRateSlider->set(0.0);
+
+        if (m_copyFrom.isEmpty()) {
+            int reset = m_pConfig->getValue<int>(
+                    ConfigKey("[Controls]", "SpeedAutoReset"), RESET_PITCH);
+            if (reset == RESET_SPEED || reset == RESET_PITCH_AND_SPEED) {
+                // Avoid resetting speed if master sync is enabled and other decks with sync enabled
+                // are playing, as this would change the speed of already playing decks.
+                if (!m_pEngineMaster->getEngineSync()->otherSyncedPlaying(getGroup())) {
+                    if (m_pRateSlider != NULL) {
+                        m_pRateSlider->set(0.0);
+                    }
                 }
             }
-        }
-        if (reset == RESET_PITCH || reset == RESET_PITCH_AND_SPEED) {
+            if (reset == RESET_PITCH || reset == RESET_PITCH_AND_SPEED) {
+                if (m_pPitchAdjust != NULL) {
+                    m_pPitchAdjust->set(0.0);
+                }
+            }
+        } else {
+            // copy rate
+            if (m_pRateSlider != NULL) {
+                m_pRateSlider->set(ControlObject::get(ConfigKey(m_copyFrom, "rate")));
+            }
+
+            // copy pitch
             if (m_pPitchAdjust != NULL) {
-                m_pPitchAdjust->set(0.0);
+                m_pPitchAdjust->set(ControlObject::get(ConfigKey(m_copyFrom, "pitch_adjust")));
+            }
+
+            // copy the play position
+            ControlObject::set(ConfigKey(getGroup(), "playposition"), ControlObject::get(ConfigKey(m_copyFrom, "playposition")));
```

Comment:
this works, but it does a standard seek which is affected by quantize. I didn't test it, but I'm guessing that will mess everything up. Also the track is slightly behind the playing track. I'll look into moving this set into the engine somehow, I still need to wrap my head around what is running in which threads and how to get the necessary data to the right place.

Reply:
That can't be the right solution. Since the play position is the position of the source track after the last engine call.
If there are engine calls in between, adopting the new Playposition at the target track it will be behind.
To avoid this you need to patch the engine code.
I think you may use an atomic varianle, or an atomic struct using ControlValueAtomic
That stores tells the engine:
"Copy the source Playposition"
"Adjust it by one buffer size in case the destination deck is has a higher index"
"Check these buffer in the destination track EngineBuffer" 
"Adopt it without quantize"

The trigger can be set here.
The actual copy needs to be done in the engine thread, in the EngineBuffer::process() call.

Reply:
I'll look into it. I think I know how to make it work, I just need to figure out how to get the play position from one enginebuffer to the other.